### DISCUSSION TITLE
[BUG] :: PnP 2.3.0 Host not reachable if VPN is used

### DISCUSSION THREAD

**Discussion by stefknoe** [pnp/powershell#3769] [Q&A]
### Reporting an Issue or Missing Feature
After updating to PnP.PowerShell 2.3.0 we get the error message "host not reachable" when trying to connect via the "Connect-PnPOnline" cmdlt. It has worked with PnP.PowerShell 2.2.0.
In our special case we use a VPN connection to connect to our company network. As long as the VPN connection is enabled "host not reachable" is thrown. If we disconnect from VPN the "Connect-PnPOnline" cmdlt works smoothly and we can connect to our tenant.

### Expected behavior 
Connect-PnPOnline should also work in case a VPN connection is used.

### Actual behavior 
**If VPN connection is enabled we see two IPv4 adresses (starting with 10 is the VPN address). With this setting the "Connect-PnPOnline" cmdlt fails with "host not reachable"**

`PS C:\Users\****> [System.Net.Dns]::GetHostAddresses('******').IPAddressToString
10.***.***.*
192.***.***.*
fe80::****
fe80::****
2001:****
2001:****
fd00::****
fd00::****
PS C:\Users\****> Connect-PnPOnline -Url "https://*****.sharepoint.com/sites/APITests" -Interactive
Connect-PnPOnline: Host not reachable`

![2024-02-20_09-56](https://github.com/pnp/powershell/assets/131651284/a214d4b7-55d1-434b-84f2-92c11721e1a2)

**If VPN is disabled the connection to our tenant can be established. There's only one IPv4 address used.**

`PS C:\Users\****> [System.Net.Dns]::GetHostAddresses('************').IPAddressToString
192.168.***.*
fe80::*****
2001:****
2001:****
fd00::****
fd00::****
PS C:\Users\****> Connect-PnPOnline -Url "https://******.sharepoint.com/sites/APITests" -Interactive
PS C:\Users\****> Get-PnPTenantSite
Url                                                                  Template             LocaleId                                                               
https://*****-my.sharepoint.com/                  SPSMSITEHOST#0       1033
https://*****.sharepoint.com/                         SitePagePublishing#0 1033
https://******.sharepoint.com/search             SRCHCEN#0            1033`

![2024-02-20_10-02](https://github.com/pnp/powershell/assets/131651284/9b805f15-ebd3-4d0a-b81e-57f647ecb7e9)

### Steps to reproduce behavior
Use the "Connect-PnPOnline" cmdlt in connection with VPN. 

### What is the version of the Cmdlet module you are running?
PnP.PowerShell 2.3.0

### Which operating system/environment are you running PnP PowerShell on?
- [x] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by GumbyKen** [comment]
I am encountering the same issue.
When I am connected to my company network with VPN and version 2.2 of PnP.PowerShell installed, I am able to successfully connect to a SharePoint site using Connect-PnPOnline, but when version 2.3 of PnP.PowerShell is installed, Connect-PnPOnline ends with a "Host not reachable" error.
When I disconnect from VPN, Connect-PnPOnline is successful with both versions of PnP.PowerShell.

However, I also get the "Host not reachable" error from version 2.3 of PnP.PowerShell when I execute Connect-PnPOnline on a machine that is directly connected to my company network.

**Comment by valpvt** [comment]
Same here, I get the issue on PnP.PowerShell 2.3.0 and now also with 2.4.0. I don't know if it's the same context but my company uses ZScaler as proxy and I suspect it to be the cause of the "Host not reachable" issue

**Reply by valpvt** [reply]
I found an update in the code that was not present in version 2.2.0. It is located in [ConnectOnline.cs](https://github.com/pnp/powershell/blob/dev/src/Commands/Base/ConnectOnline.cs), line 784.
The following code does not work behind corporate proxy `var conn = System.Net.Dns.GetHostEntry(nameOrAddress);`

**Comment by mrupaner** [comment]
I'm having the same issue for versions 2.3.0 and 2.4.0. At proxy level (ZScaler), we just found allowed traffic 200 http code.
